src/libultra_code/pfsreadwritefile.c

```diff
@@ -0,0 +1,139 @@
+#include <ultra64.h>
+#include <global.h>
+
+#define CHECK_IPAGE(p, pfs)                                                                                   \
+    (((p).ipage >= (pfs).inodeStartPage) && ((p).inode_t.bank < (pfs).banks) && ((p).inode_t.page >= 0x01) && \
+     ((p).inode_t.page < 0x80))
+
```

Comment:
Same thing for this macro, it's already in a header so I don't think it has to be defined again.
```suggestion
```

Reply:
There are two different CHECK_IPAGE macros. One has only the inode unit as an argument, and the other has the inodeu nit and the pfs as an argument. That's why I had the macros in C files (accidentally had one of the macros in a header file). One version of the macro is used in pfsreadwritefile.c, and the other version is used in pfschecker.c. But it doesnt look like they are used anywhere else.